src/main/java/io/micronaut/build/MicronautQualityChecksAggregatorPlugin.java

```diff
@@ -6,48 +6,43 @@
 import org.gradle.api.artifacts.Configuration;
 import org.gradle.api.attributes.TestSuiteType;
 import org.gradle.api.plugins.BasePlugin;
-import org.gradle.api.plugins.JavaPlugin;
 import org.gradle.api.reporting.ReportingExtension;
 import org.gradle.testing.jacoco.plugins.JacocoCoverageReport;
-import org.gradle.testing.jacoco.plugins.JacocoPlugin;
 import org.gradle.testing.jacoco.plugins.JacocoReportAggregationPlugin;
 import org.gradle.testing.jacoco.tasks.JacocoReport;
 import org.sonarqube.gradle.SonarQubeExtension;
 import org.sonarqube.gradle.SonarQubePlugin;
 
-import java.util.Collections;
-
 @SuppressWarnings("UnstableApiUsage")
-public class MicronautQualityChecksPlugin implements Plugin<Project> {
+public class MicronautQualityChecksAggregatorPlugin implements Plugin<Project> {
 
     public static final String COVERAGE_REPORT_TASK_NAME = "testCodeCoverageReport";
-    public static final String TEST_REPORT_TASK_NAME = "jacocoTestReport";
 
     @Override
     public void apply(final Project rootProject) {
         if (!rootProject.equals(rootProject.getRootProject())) {
-            throw new GradleException("The Quality Checks plugin must only be applied to the root project");
+            throw new GradleException("The Quality Checks Aggregator plugin must only be applied to the root project");
         }
 
         configureSonar(rootProject);
         configureJacoco(rootProject);
-        rootProject.getSubprojects().forEach(this::subprojectApply);
     }
 
-    private void subprojectApply(final Project subproject) {
-        configureJacocoForSubproject(subproject);
-    }
 
     private void configureSonar(final Project rootProject) {
         rootProject.getPluginManager().apply(SonarQubePlugin.class);
         final SonarQubeExtension sonarQubeExtension = rootProject.getExtensions().findByType(SonarQubeExtension.class);
         final String githubSlug = (String) rootProject.findProperty("githubSlug");
-        sonarQubeExtension.properties(p -> {
-            p.property("sonar.projectKey", githubSlug.replaceAll("/", "_"));
-            p.property("sonar.organization", "micronaut-projects");
-            p.property("sonar.host.url", "https://sonarcloud.io");
-            p.property("sonar.java.source", "8");
-        });
+        if (sonarQubeExtension != null) {
+            sonarQubeExtension.properties(p -> {
+                if (githubSlug != null) {
+                    p.property("sonar.projectKey", githubSlug.replaceAll("/", "_"));
+                    p.property("sonar.organization", "micronaut-projects");
+                    p.property("sonar.host.url", "https://sonarcloud.io");
+                    p.property("sonar.java.source", "8");
```

Comment:
Note for later: we have several places where 8 is hardcoded, including in build files themselves. We should move away from `targetCompatibility = 8` to something which uses the `micronautBuild.languageLevel` property.